sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,445 @@
+    }
+
+    /**
+     * Adds a set tier operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobBatch.setTier#String-AccessTier-LeaseAccessConditions}
+     *
+     * @param blobUrl URI of the blob.
+     * @param accessTier The tier to set on the blob.
+     * @param leaseAccessConditions Lease access conditions that must be met to allow this operation.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> setTier(String blobUrl, AccessTier accessTier, LeaseAccessConditions leaseAccessConditions) {
+        return setTierHelper(getUrlPath(blobUrl), accessTier, leaseAccessConditions);
+    }
+
+    private Response<Void> setTierHelper(String urlPath, AccessTier accessTier,
+        LeaseAccessConditions leaseAccessConditions) {
+        setBatchType(BlobBatchType.SET_TIER);
+        return createBatchOperation(batchClient.setAccessTierWithResponse(accessTier, null, leaseAccessConditions),
+            urlPath, EXPECTED_SET_TIER_STATUS_CODES);
+    }
+
+    private <T> Response<T> createBatchOperation(Mono<Response<T>> response, String urlPath,
+        int... expectedStatusCodes) {
+        int id = contentId.getAndIncrement();
+        batchOperationQueue.add(response
+            .subscriberContext(Context.of(BATCH_REQUEST_CONTENT_ID, id, BATCH_REQUEST_URL_PATH, urlPath)));
+
+        BlobBatchOperationResponse<T> batchOperationResponse = new BlobBatchOperationResponse<>(expectedStatusCodes);
+        batchMapping.put(id, batchOperationResponse);
+        return batchOperationResponse;
+    }
+
+    private String getUrlPath(String url) {
+        return UrlBuilder.parse(url).getPath();
+    }
+
+    private void setBatchType(BlobBatchType batchType) {
+        if (this.batchType == null) {
+            this.batchType = batchType;
+        } else if (this.batchType != batchType) {
+            throw logger.logExceptionAsError(new UnsupportedOperationException(String.format(Locale.ROOT,
+                "'BlobBatch' only supports homogeneous operations and is a %s batch.", this.batchType)));
+        }
+    }
+
+    Flux<ByteBuffer> getBody() {
+        if (batchOperationQueue.isEmpty()) {
+            throw logger.logExceptionAsError(new UnsupportedOperationException("Empty batch requests aren't allowed."));
+        }
+
+        Disposable disposable = Flux.fromStream(batchOperationQueue.stream())
+            .flatMap(batchOperation -> batchOperation)
+            .subscribe();
+
+        while (!disposable.isDisposed()) {
+            // Wait until the batch operation has processed in the pipeline.
+            // This is used as opposed to block as it won't trigger an exception if ran in a Reactor thread.
+        }
```

Comment:
Think I left a comment on the other PR, but this I think can just be .blockLast or equivalent

Reply:
It could but I went with this approach as it is safer for use inside Reactor threads, they will throw an exception by default if a blocking call is made in them.

Reply:
XD you sneaky mom!

---
In reply to: [333121225](https://github.com/Azure/azure-sdk-for-java/pull/5734#discussion_r333121225) [](ancestors = 333121225)